src/pybind/mgr/dashboard/frontend/src/app/shared/directives/tab-activate-by-fragment.directive.ts

```diff
@@ -0,0 +1,35 @@
+import { Directive, Host, HostListener, Input, OnInit, Optional } from '@angular/core';
+import { Router, UrlTree } from '@angular/router';
+
+import * as _ from 'lodash';
+import { TabDirective } from 'ngx-bootstrap/tabs';
+
+/**
+ * This directive will activate the tab if the route fragment is
+ * identical to its identifier. It will also navigate to the current
+ * route when the tab is selected by adding the tab identifier as
+ * fragment to the current route, e.g. `/monitoring#silences`.
+ */
+@Directive({
+  selector: '[cdTabActivateByFragment]'
+})
+export class TabActivateByFragmentDirective implements OnInit {
+  constructor(@Optional() @Host() private tab: TabDirective, private router: Router) {}
+
+  @Input()
+  fragment: string;
+
+  ngOnInit() {
+    const urlTree: UrlTree = this.router.parseUrl(this.router.url);
+    if (urlTree.fragment === this.fragment) {
+      _.set(this.tab, 'active', true);
```

Comment:
Shouldn't we set also the `id`? In this it works great because the tabs are placed in the top of the page, but if not, the browser won't scroll to the "#" anchor.